### PR TITLE
TINY-6728: Redirect content apis to rtc plugin

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket:

Description of Changes:
* This moves the content api overrides for rtc into the rtc plugin it also namespaces the rtc api since it getting hard to know what is what and name clashes might start occuring.

* [x] The rockfest PR needs to be merged at the same time as this one since this is breaking the api contract between the two parts.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] License headers added on new files (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/Rtc.ts [hunk 1]
```diff
@@ -190,89 +167,38 @@ const makeRtcAdaptor = (tinymceEditor: Editor, rtcEditor: RtcRuntimeApi): RtcAda
...
-      reset: () => rtcEditor.reset(),
-      hasUndo: () => rtcEditor.hasUndo(),
-      hasRedo: () => rtcEditor.hasRedo(),
-      transact: (_undoManager, _locks, fn) => {
-        rtcEditor.transact(fn);
-        return createDummyUndoLevel();
-      },
-      ignore: (_locks, callback) => rtcEditor.ignore(callback),
-      extra: (_undoManager, _index, callback1, callback2) => rtcEditor.extra(callback1, callback2)
+      undo: () => rtcEditor.undoManager.undo(),
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/Rtc.ts:170]: There's a lot of duplication here that won't get minified, so should we looking at breaking that out? eg:
```
const undoManger = rtcEditor.undoManager;
const formatter = rtcEditor.formatter;
const editor = rtcEditor.editor;
....
```

**Review by lnewson — COMMENTED** [review]
Looks okay to me, but not approving yet because of the question and because we need to fix the deepscan error as well first. eg
```
const makeRtcAdaptor = (tinymceEditor: Editor, rtcEditor: RtcRuntimeApi): RtcAdaptor => {
```
should probably become something like:
```
const makeRtcAdaptor = (rtcEditor: RtcRuntimeApi): RtcAdaptor => {
```

File: modules/tinymce/src/core/main/ts/Rtc.ts (hunk 1 above)
Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/Rtc.ts:170]: I've been wondering whether we should just implement the entire undoManager etc api in RTC. I'm not sure what's left that we aren't overriding.

Comment (spocke) [reply modules/tinymce/src/core/main/ts/Rtc.ts:170]: There are some minor things like a data array etc. But I prefer us having the same format on these overrides it is also now very easy to read. In theory we could have some form of make undoManager thing in the rtc that returns a whole object and just replaces the instance or something but I think that would be a bit more messy and not as clear as this.

**Review by lnewson — APPROVED** [review]

**Review by metricjs — APPROVED** [review]

**Pull request merged**